r-package/grf/R/causal_forest.R

```diff
@@ -216,36 +216,42 @@ causal_forest <- function(X, Y, W,
   } else if (length(W.hat) != nrow(X)) {
     stop("W.hat has incorrect length.")
   }
-
+  pre.tuning.parameters <- c(
+    min.node.size = validate_min_node_size(min.node.size),
+    sample.fraction = validate_sample_fraction(sample.fraction),
+    mtry = validate_mtry(mtry, X),
+    alpha = validate_alpha(alpha),
+    imbalance.penalty = validate_imbalance_penalty(imbalance.penalty))
   if (tune.parameters) {
-    tuning.output <- tune_causal_forest(X, Y, W, Y.hat, W.hat,
-      sample.weights = sample.weights,
-      num.fit.trees = num.fit.trees,
-      num.fit.reps = num.fit.reps,
-      num.optimize.reps = num.optimize.reps,
-      min.node.size = min.node.size,
-      sample.fraction = sample.fraction,
-      mtry = mtry,
-      alpha = alpha,
-      imbalance.penalty = imbalance.penalty,
-      stabilize.splits = stabilize.splits,
-      num.threads = num.threads,
-      honesty = honesty,
-      honesty.fraction = honesty.fraction,
-      prune.empty.leaves = prune.empty.leaves,
-      seed = seed,
-      clusters = clusters,
-      samples.per.cluster = samples.per.cluster
-    )
+    tuning.output <- tryCatch({
```

Comment:
The tryCatch statement should be at the most granular level possible. So for example if there's a risky call to DiceKriging, we should surround that in a tryCatch, but nothing else.

My intuition is that we should keep this part of the code roughly as it was before, and put all the error checking inside `tune_causal_forest` (so that `tune_causal_forest` is smart enough to always return an answer, and to return default parameter values if it failed to run DiceKriging).

Reply:
That makes a lot of sense. Done!